### DISCUSSION TITLE
Making dynamic zone component content an object

### DISCUSSION THREAD

**Discussion by lucaslevin** [strapi/strapi#6596] [Legacy Topics]
I have a single type of Home and in that I have a dynamic zone with a hero section. When getting the data I get:

`{
id: 2,
created_at: "2020-06-08T22:30:00.876Z",
updated_at: "2020-06-08T22:30:00.876Z",
sections: [
{
__component: "page.hero",
id: 8,
title: null,
description: null
}
]
}`

I would like the title and description (the content of my hero) to be in an object like "content". So like this:

`{
__component: "page.hero",
id: 8,
content: { title: null,
description: null }
}`

I tried making a new component with "content" and added title and description. But my content will change values, so that doesn't work for me.

This is my home.settings.json:

`{
  "kind": "singleType",
  "collectionName": "homes",
  "info": {
    "name": "home"
  },
  "options": {
    "increments": true,
    "timestamps": true
  },
  "attributes": {
    "sections": {
      "type": "dynamiczone",
      "components": ["page.text-block", "page.hero"]
    }
  }
}
`

And my hero.json:

`{
  "collectionName": "components_page_heroes",
  "info": {
    "name": "hero",
    "icon": "bullseye"
  },
  "options": {},
  "attributes": {
    "title": {
      "type": "string"
    },
    "description": {
      "type": "richtext"
    }
  }
}
`

I'm pretty new at this, but hoping to get some help :-)

**Comment by derrickmehaffy** [comment]
When you created your component did you select single or repeatable?
